addons/udes_stock/models/models.py

```diff
@@ -45,3 +48,48 @@ def base_model_detailed_view(self, model, form_view):
             "res_id": self.id,
             "context": {"view_all_fields": True},
         }
+
+    @api.model
+    def _create(self, vals):
+        """Inheriting low level _create method to check if there is any date or datetime field less
+        than a static year 1000. In that case raise an error to show the users that the date entered
+         is not correct"""
+        date_fields = self.get_date_or_datetime_fields()
+        if isinstance(vals, dict):
+            self.validate_values_of_date_fields(vals, date_fields)
+        res = super()._create(vals)
+        return res
+
+    @api.multi
+    def _write(self, vals):
+        """Inheriting low level _write method to check if there is any date or datetime field less
+        than a static year 1000. In that case raise an error to show the users that the date entered
+         is not correct"""
+
+        date_fields = self.get_date_or_datetime_fields()
+        if isinstance(vals, dict):
+            self.validate_values_of_date_fields(vals, date_fields)
+        res = super()._write(vals)
+        return res
+
+    @api.model
+    def validate_values_of_date_fields(self, values, date_fields):
+        """
+        Checking if date or datetime fields that are formatted as string
+        can be converted to a date field and the year is greater than 1000.
+        Raising error in case year is less than 1000
+        """
+        for key, value in values.items():
```

Comment:
Suggestion: `date_values = {k: v for k, v in values.items() if k in date_fields and isinstance(v, str) and len(v) >= 10}`

Then can just go through the values in `date_values`.